src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +31,263 @@ public void GetSerializableMembers_Interface()
+        {
+            yield return new object[] { typeof(int[]) };
+            yield return new object[] { typeof(int[,]) };
+            yield return new object[] { typeof(int).MakePointerType() };
+            yield return new object[] { typeof(int).MakeByRefType() };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_NotSupportedType_TestData))]
+        public void GetUninitializedObject_NotSupportedType_ThrowsArgumentException(Type type)
+        {
+            Assert.Throws<ArgumentException>(null, () => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<ArgumentException>(null, () => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [InlineData(typeof(AbstractClass))]
+        [InlineData(typeof(StaticClass))]
+        [InlineData(typeof(Interface))]
+        [InlineData(typeof(Array))]
+        public void GetUninitializedObject_AbstractClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        private abstract class AbstractClass { }
+        private static class StaticClass { }
+        private interface Interface { }
+
+        public static IEnumerable<object[]> GetUninitializedObject_OpenGenericClass_TestData()
+        {
+            yield return new object[] { typeof(GenericClass<>) };
+            yield return new object[] { typeof(GenericClass<>).MakeGenericType(typeof(GenericClass<>)) };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_OpenGenericClass_TestData))]
+        public void GetUninitializedObject_OpenGenericClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        public interface IGenericClass
+        {
+            int Value { get; set; }
+        }
+
+        public class GenericClass<T> : IGenericClass
+        {
+            public int Value { get; set; }
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_TestData()
+        {
+            yield return new object[] { typeof(ArgIterator) };
+            yield return new object[] { typeof(RuntimeArgumentHandle) };
+            yield return new object[] { typeof(TypedReference) };
+
+            yield return new object[] { typeof(Span<int>) };
+            yield return new object[] { typeof(ReadOnlySpan<int>) };
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_NetCore_TestData()
+        {
+            yield return new object[] { Type.GetType("System.ByReference`1[System.Int32]") };
```

Comment:
Reflection on ByReference will be blocked on uapaot and I'm not sure what corert will do. @jkotas , do you have any ideas on how to get a byref-like Type on those runtimes?